lib/main.js

```diff
@@ -259,10 +339,14 @@ function selectCert(userCerts, text) {
                                     "_blank",
                                     "dialog,centerscreen,chrome,modal",
                                     sandbox);
+  let result = {
+    nickname: sandbox.selectedCert,
+    password: sandbox.certPassword
+  };
   if (sandbox.cancelled) {
-    return ERROR_USER_CANCEL;
+    result.nickname = ERROR_USER_CANCEL;
```

Comment:
Since result is now a more complex type, let's just add an 'error' or 'status' field to it. If it's empty, it means no error. If it is set, there was an error (ERROR_USER_CANCEL in this case).